h5netcdf/core.py

```diff
@@ -366,30 +368,50 @@ def _ensure_dim_id(self):
                 self._h5ds.attrs["_Netcdf4Dimid"] = dim.attrs["_Netcdf4Dimid"]
 
     def _maybe_resize_dimensions(self, key, value):
-        """Resize according to given (expanded) key with respect to variable dimensions"""
+        """Resize according to given (expanded) key with respect to variable dimensions.
+
+        Parameters
+        ----------
+        key : Tuple[slice]
+            Indexing key
+        value : array-like
+            Values to be written.
+        """
         new_shape = ()
-        v = None
+        v = np.asarray(value)
         for i, dim in enumerate(self.dimensions):
             # is unlimited dimensions (check in all dimensions)
             if self._parent._all_dimensions[dim].isunlimited():
-                if key[i].stop is None:
+                current_dim_size = len(self._parent._all_dimensions[dim])
+                if (
+                    key[i].stop is None
+                ):  # TODO: Won't this fail if key[i] is an array of ints?
```

Comment:
No, at this point in time we have key normalized to slices already.

```suggestion
                ):
```